tests/ResourceSyncingTest.php

```diff
@@ -126,6 +126,34 @@
     ], ResourceUser::first()->getAttributes());
 });
 
+test('creating the resource in tenant database creates it in central database and used custom attributes', function () {
+    // Assert no user in central DB
+    expect(ResourceUser::all())->toHaveCount(0);
+
+    $tenant = ResourceTenant::create();
+    pest()->artisan('tenants:migrate', [
+        '--path' => __DIR__ . '/Etc/synced_resource_migrations/custom',
+        '--realpath' => true,
+    ])->assertExitCode(0);
+
+    tenancy()->initialize($tenant);
+
+    // Create the same user in tenant DB
+    ResourceUser::create([
+        'global_id' => 'acme',
+        'name' => 'John Doe',
+        'email' => 'john@localhost',
+        'password' => 'secret',
+        'role' => 'commenter', // unsynced
+        'code' => 'bar' // extra column which does not exist in central users table
+    ]);
+
+    tenancy()->end();
+
+    // assert user was created
+    expect(CentralUser::first()->global_id)->toBe('acme');
```

Comment:
I'd make a specific assertion about the `code`:
```suggestion
        'code' => 'bar' // extra column which does not exist in central users table
    ]);

    tenancy()->end();

    // Assert central user was created without `code` property
    expect(CentralUser::first()->global_id)->toBe('acme');
    expect(CentralUser::first()->code)->toBeNull();
```
Also, does the `role` exist in the central DB?

Reply:
> does the role exist in the central DB

Yes.